.github/workflows/build-docker.yml

```
@@ -0,0 +1,94 @@
+name: "Build Pipeline (Docker)"
+
+on:
+  push:
+    branches:
+      - jesse/pipeline-deploy
+      - main
+
+env:
+  IMAGE_NAME: "${{ vars.GAR_LOCATION }}-docker.pkg.dev/${{ vars.PROJECT_ID }}/${{ secrets.DOCKER_REPOSITORY }}/polygon-cli"
+
+  ATTESTOR_PROJECT_ID: ${{ vars.PROJECT_ID }}
+  KEY_RING: "jesse-artifacts-ring"
+  KEY: "jesse-artifacts-binary-auth-key"
+  ATTESTOR: "jesse-artifacts-attestor"
+
+jobs:
+  build-pipeline-docker:
+    name: "Build, push, scan, and sign Docker image"
+    permissions:
+      contents: "write"
+      id-token: "write"
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Google Auth
+        id: auth
+        uses: google-github-actions/auth@v2
+        with:
+          token_format: "access_token"
+          workload_identity_provider: "${{ secrets.WIF_PROVIDER }}" # this was hard to find: WIP --> Expand pool --> Click pencil icon"
+          service_account: "${{ secrets.WIF_SERVICE_ACCOUNT }}"
+
+      - name: Docker Auth
+        id: docker-auth
+        uses: docker/login-action@v1
+        with:
+          username: "oauth2accesstoken"
+          password: "${{ steps.auth.outputs.access_token }}"
+          registry: "${{ vars.GAR_LOCATION }}-docker.pkg.dev"
+
+      - name: Build and Push Docker Image to GCP Artifact Registry
+        run: |-
+          docker build -t "${{ env.IMAGE_NAME }}:${{ github.sha }}" .
+          docker push "${{ env.IMAGE_NAME }}:${{ github.sha }}"
+
+      - name: "Set up Cloud SDK"
+        uses: "google-github-actions/setup-gcloud@v1"
```

https://github.com/google-github-actions/setup-gcloud/releases

```suggestion
        uses: google-github-actions/setup-gcloud@v2
```